packages/headless/src/features/commerce/facets/facet-search-set/commerce-facet-search-actions.ts

```diff
@@ -0,0 +1,69 @@
+import {AsyncThunkPayloadCreator, createAsyncThunk} from '@reduxjs/toolkit';
+import {
+  CommerceAPIResponse,
+  CommerceFacetSearchAPIClient,
+} from '../../../../api/commerce/commerce-api-client';
+import {SpecificFacetSearchResponse} from '../../../../api/search/facet-search/specific-facet-search/specific-facet-search-response';
+import {AsyncThunkOptions} from '../../../../app/async-thunk-options';
+import {ClientThunkExtraArguments} from '../../../../app/thunk-extra-arguments';
+import {requiredNonEmptyString} from '../../../../utils/validate-payload';
+import {buildCommerceFacetSearchRequest} from './commerce-facet-search-request-builder';
+import {StateNeededForCommerceFacetSearch} from './commerce-facet-search-state';
+
+type ExecuteCommerceFacetSearchThunkReturn = {
+  facetId: string;
+  response: CommerceAPIResponse<SpecificFacetSearchResponse>;
+};
+
+type ExecuteCommerceFacetSearchThunkArg = string;
+
+type ExecuteCommerceFacetSearchThunkApiConfig = AsyncThunkOptions<
+  StateNeededForCommerceFacetSearch,
+  ClientThunkExtraArguments<CommerceFacetSearchAPIClient>
+>;
+
+const getExecuteFacetSearchThunkPayloadCreator =
+  (
+    isFieldSuggestionsRequest: boolean
```

Comment:
nit: using an enum instead would make calls to this method a tiny bit clearer, but I'm being nitpicky
```suggestion
    isFieldSuggestionsRequest: 'search' | 'suggestion'
```

Reply:
It's a boolean in non-commerce facet search so I'll keep it as is for the sake of consistency